mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+[P0] FalseNegatives: Calculates the number of false negatives.
+[P0] FalsePositives: Calculates the number of false positives.
+https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix
+
+[P0] Precision: Computes the precision of the predictions with respect to the labels.
+[P0] Recall: Computes the recall of the predictions with respect to the labels.
+[P0] AUC: Approximates the AUC (Area under the curve) of the ROC or PR curves.
+[P0] F1 Score: 2*precision*recall / (precision+recall)
+
+[P0] BinaryClassConfusionMatrix
+
+Plots
+[P0] Confusion matrix
+[P0] Interactive ROC curve with metrics (TP/TN/FP/FN/Acc/F1/AUC), binary classification
+[P0] Lift chart
+
+Global explainability
+[P0] Model built-in feature importance (supported models)
+[P0] SHAP explainers
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+
+    @classmethod
+    def save_content_to_file(cls, content, output_artifact_path):
+        assert isinstance(content, Image)
+        content.save(output_artifact_path)
+
+    @classmethod
+    def load_content_from_file(cls, local_artifact_path):
+        return open_image(local_artifact_path)
+
+
+class DefaultEvaluator(ModelEvaluator):
+
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_image_artifact(self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name):
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        mlflow.log_artifact(artifact_file_local_path, artifact_file_name)
+        artifacts[artifact_file_name] = \
+            ImageEvaluationArtifact(
+                location=get_artifact_uri(run_id, artifact_file_name),
+                content=ImageEvaluationArtifact.load_content_from_file(artifact_file_local_path)
+            )
+
+    def _log_model_explainality(
+        self, artifacts, temp_dir, model, X, dataset_name, run_id
+    ):
+        explainer = shap.Explainer(model, X)
+        shap_values = explainer(X)
+
+        def plot_summary():
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(artifacts, temp_dir, plot_summary, run_id, 'shap_summary', dataset_name)
+
+        def plot_feature_importance():
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_feature_importance, run_id, 'shap_feature_importance', dataset_name
+        )
+
+    def _classifier_compute_metrics_and_compute_and_log_artifacts(
+        self, temp_dir, model, X, y, dataset_name, run_id, evaluator_config
+    ):
+        # TODO: how to get num_classes without extra config ?
```

Comment:
groupby the label/target and do a count of the groups on the training data

Reply:
Here we can only get the evaluation dataset.

Reply:
My thoughts on this:
The only sure way to know what the cardinality is of a classification model in a generic manner (not digging in to inspect attributes that may or may not be present in the model object) is to record this data at time of training. Storing a count of distinct values of the target series may be the best option here. 
Although, this begs the question of whether we want to store additional information on the model type as some sort of Enumeration value for selecting suites of metrics to evaluate as defined at training time based on the model type that is selected for training. 
Purely as an example here for us to discuss further as a team: 

Binary Classifier
Multiclass Classifier
LinearRegressor*
TreeRegressor*
Unsupervised
Statistical
Neural

* the reason for splitting these out is because SHAP has different solvers that are non-mutually exclusive and a tree-solver SHAP estimator is going to run an order of magnitude faster than a brute-force kernel solver, while a linear solver works best for linear MLE-optimized models. Kernel solver is the 'catch all' but it runs painfully slow and runs through a different underlying implementation that actually gives different responses than the more 'native' ones. 
* For explainability reasons and the efficiency differences, having some key critical data about 'what was the nature of this model' without having to maintain a great deal of mapping logic for each individual model type might be useful for the remainder of the implementation.

Reply:
Current code calculate cardinality on evaluation dataset labels as num_classes. 
We can improve this in follow-up task.

Reply:
SGTM